src/components/carousel/index.js

```diff
@@ -0,0 +1,102 @@
+import React, { useEffect, useRef, useState } from "react";
+import PropTypes from 'prop-types'
+import { useWindowWidth } from "../../hooks";
+import { panelType } from './types'
+import { CarouselPanel } from './panels'
+import { PauseIcon, PlayIcon } from "../icons";
+import { Overlay, Border, Wrapper, StyledPanelWrapper } from './panels/subcomponents/PanelContainer'
+import { useTransition } from "react-spring";
+import { StateNote } from './panels/subcomponents/StateNote'
+
+const INTERVAL = 5000 // ms
+
+export const Carousel = ({ panels }) => {
+  const { isCompact } = useWindowWidth();
+  const [carouselIndex, setCarouselIndex] = useState(0)
+  const indexRef = useRef(carouselIndex);
+  indexRef.current = carouselIndex;
+
+  // pause/play functionality is controlled by this playingAnimations state variable
+  // when playingAnimations is set to false, the carousel does not advance to the next panel
+  // see documentation here: https://www.w3.org/WAI/WCAG21/Understanding/pause-stop-hide.html
+  const [playingAnimations, setPlayingAnimations] = useState(true);
+
+  // this fires when the index changes and playingAnimations changes
+  useEffect(() => {
+
+    // if playing, wait for INTERVAL ms before advancing to the next panel
+    if (playingAnimations) {
+      let timer;
+      timer = setInterval(
+        () => setCarouselIndex((carouselIndex + 1) % panels.length),
+        INTERVAL,
+      )
+      return () => clearInterval(timer)
+    }
+  }, [carouselIndex, panels, playingAnimations])
+
+  const handleKeyDown = event => {
+    if (event.keyCode === 32) {
+      event.preventDefault();
+      setPlayingAnimations(!playingAnimations);
+    }
+  };
+  
+  // panelTransitions is an array created using the useTransition hook from react-spring
+  // This array adds transition variables and is mapped over  inside the Overlay component. 
+  // It takes three arguments, 1- the data that you want to display, 2- a key, 3- the config object
+  // The component that is returned is a styled component with a base of animated.div called StyledPanelWrapper
+  // Inside StyledPanelWrapper is the CarouselPanel component, which displays only the current panel
+  // Useful Links: https://react-spring.dev/docs/components/use-transition
+  // Useful Video Walkthrough: https://www.youtube.com/watch?v=WKmhhBokAh8
+
+  const panelTransitions = useTransition(
```

Comment:
just a thought here: the horizontal entering/exiting of panels is nice on desktop but feels less streamlined on small displays because, i think, the side edges of that darker block are visible. what do you think about a vertical transition for small displays?